### ISSUE TITLE
Request: Option to merge/batch function signatures and emit them in a single extern "C" block.

### ISSUE THREAD

**Issue by Lokathor** [rust-lang/rust-bindgen#1742]
Instead of code like
```rust
extern "C" {
  pub fn SDL_LockAudio();
}
extern "C" {
  pub fn SDL_LockAudioDevice(dev: SDL_AudioDeviceID);
}
extern "C" {
  pub fn SDL_UnlockAudio();
}
```

Emit something like
```rust
extern "C" {
  pub fn SDL_LockAudio();
  pub fn SDL_LockAudioDevice(dev: SDL_AudioDeviceID);
  pub fn SDL_UnlockAudio();
}
```

(I'm not sure if I've requested this before but I didn't see it in the tracker)

**Comment by emilio** [comment]
There's previous requests like this in #1261. The somewhat complicated part is that since we don't have any ordering guarantees, each item needs to be able to generate itself. We could try to merge adjacent items or such but that'd be non-obvious.

I think it'd be less-hackily-solvable with something like #1743 or such

**Comment by Lokathor** [comment]
Did you mean to close this? It seems like it might be blocked on #1743 but once that is done i think this would be a separate issue.

**Comment by emilio** [comment]
Yeah, I intended to close it but because https://github.com/rust-lang/rust-bindgen/issues/564 is a dupe of this, sorry, should've pointed it out.
